tensorflow-opgen/src/TensorFlow/OpGen/ParsedOp.hs

```diff
@@ -104,6 +103,13 @@ data AttrBaseType = AttrBytes | AttrInt64 | AttrFloat | AttrBool
                 | AttrType | AttrShape | AttrTensor
                 deriving Eq
 
+data TypeParam = TypeParam
+    { typeParamIsList :: Bool
+    , typeParamRestrictions :: Maybe [DataType]
+        -- ^ The list of allowed types (see: TensorFlow.Types.OneOf).
+        -- If 'Nothing', then any type is acceptable.
```

Comment:
What does Just [] indicate? Does it make sense to drop Maybe and use [] as anything-goes value? Alternatively, Maybe NonEmpty would work.